ceph/ceph_admin/cephadm.py

```diff
@@ -0,0 +1,200 @@
+
+        cmd.append("--")
+        cmd.extend(args)
+
+        out = self.installer.exec_command(
+            sudo=True,
+            cmd=" ".join(cmd),
+            timeout=timeout,
+            check_ec=check_status,
+            long_running=long_running,
+        )
+
+        if isinstance(out, tuple):
+            log.debug(out[0])
+
+        return out
+
+    def _get_dir_list(self, dir_path, role=None):
+        """Get dir list from dir_path"""
+        dir_list = {}
+        if role and isinstance(role, str):
+            role = [str(role)]
+
+        for node in self.nodes:
+            if role and node.role not in role:
+                continue
+
+            dir_list[node.shortname] = node.get_dir_list(sudo=True, dir_path=dir_path)
+
+        return dir_list
+
+    def get_ceph_config_dir(self, role=["mon", "osd"]):
+        """Get file/dir lists from '/etc/ceph'"""
+        return self._get_dir_list(dir_path=CEPH_CONFIG_DIR, role=role)
+
+    def get_ceph_lib_dir(self, role=["mon", "osd"]):
+        """Get file/dir lists from '/var/lib/ceph'"""
+        return self._get_dir_list(dir_path=CEPH_LIB_DIR, role=role)
+
+    def get_cephadm_lib_dir(self, role=None):
+        """Get file/dir lists from '/var/lib/cephadm'"""
+        return self._get_dir_list(dir_path=CEPHADM_LIB_DIR, role=role)
+
+    def _get_running_containers(self, expr=None, role=None):
+        """Get containers running on nodes"""
+        cmd = "podman ps --noheading"
+        if filter:
+            cmd += f" --filter {expr}"
+
+        if role and isinstance(role, str):
+            role = [str(role)]
+
+        containers_list = {}
+        for node in self.nodes:
+            if role and node.role not in role:
+                continue
+
+            containers_list[node.shortname], _ = node.exec_command(sudo=True, cmd=cmd)
+
+        return containers_list
+
+    def get_running_ceph_containers(self, role=None):
+        """Get containers running on nodes"""
+        return self._get_running_containers(expr="name=ceph-*", role=role)
+
+    def get_ceph_disks(self):
+        """Get disks used by ceph cluster"""
+        cmd = "lsblk -ln -o name | grep ceph-"
+        disks_list = {}
+        for node in self.nodes:
+            disks_list[node.shortname], _ = node.exec_command(
+                sudo=True, cmd=cmd, check_ec=False
+            )
+
+        return disks_list
+
+    def get_ceph_status(self):
+        """Get ceph cluster status"""
+        out, _ = self.shell(args=["ceph", "-s"])
+        if out:
+            log.info(f"Ceph cluster status '{out}'")
+            return out
+
+    def get_cluster_fsid(self):
+        """Get ceph cluster FSID.
+
+        Returns:
+           str : Ceph cluster FSID
+        """
+        out, _ = self.shell(args=["ceph", "fsid"])
+        if out:
+            log.info(f"Cluster FSID '{out}'")
+            return out.strip()
+
+    def disable_module(self, module):
```

Comment:
IMO, we could move this method to `mgr` module

Reply:
Done